src/components/Input.js

```diff
@@ -3,14 +3,29 @@ import PropTypes from 'prop-types';
 import classnames from 'classnames';
 import defaultTheme from '../themes/DefaultTheme';
 import darkTheme from '../themes/DarkTheme';
-import { ErrorIcon } from './icons';
+import { PublicIcon, PrivateIcon, ErrorIcon } from './icons';
 
 class Input extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      type: props.type,
```

Comment:
I would suggest to change that state property to `passwordVisible: props.passwordVisible` and use it to indicate if the password should be visible or not by default. This will let us to create password inputs where the password will be visible by default ;)